packages/notebook-extension/src/tool-widgets/metadataEditorFields.tsx

```diff
@@ -61,6 +61,12 @@ export class CellMetadataField extends NotebookTools.MetadataEditorTool {
 
   private _onSourceChanged() {
     if (this.editor.source) {
+      const metadataKeys = Object.keys(
+        this._tracker.activeCell?.model.sharedModel.metadata ?? {}
+      );
+      for (const key of metadataKeys) {
+        this._tracker.activeCell?.model.sharedModel.deleteMetadata(key);
+      }
```

Comment:
Just by looking at the code, I have a feeling that lines 67-72 may need to be wrapped in a single `transaction` as for the user this should be undoable as a single change.

For reference see:

https://github.com/jupyterlab/jupyterlab/blob/07bef086048a3ca995db347537bf25726ddc2168/packages/notebook/src/actions.tsx#L223-L226